spec/api/v1/users_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe V1::Users do
+  let(:api_token) { create :api_token }
+
+  describe '#GET /api/v1/users' do
+
+    context 'when api token is invalid' do
+      let(:api_token) { double :api_token, token: 'foo' }
+
+      before do
+        get '/api/v1/users', api_key: api_token.token
+      end
+
+      it 'returns a authorization error' do
+        expect(response.body).to match(/Invalid token/)
+      end
+    end
+
+    context 'when api token is linked with a team' do
+      let(:some_team) { create :team }
+      let(:api_token) { create :api_token, team: some_team }
+    
+      before do
+        user1 = create :user, name: 'new_user_1', created_at: 1.hour.ago
+        user2 = create :user, name: 'new_user_2', created_at: 1.hour.ago
+
+        some_team.users << [user1]
+
+        get '/api/v1/users', api_key: api_token.token
+      end
+
+      it 'returns only the users associated with current team' do
+        user_name = JSON.parse(response.body).map{ |u| u['name'] } 
+        expect(user_name).to contain_exactly('new_user_1')
+      end
+    end
+
+    context 'when fetch user' do
+      before do
+        create_list :user, 4
+
+        get '/api/v1/users', per_page: 2, api_key: api_token.token
+      end
+
+      it 'returns 2 users' do
+        expect(JSON.parse(response.body).count).to eq(2)
+      end
+    end
+
+    context 'when fetch only new users' do
+      before do
+        create :user, name: 'new_user_1', created_at: 2.days.ago
+        create :user, name: 'new_user_2', created_at: 3.days.ago
+        create :user, name: 'new_user_3', created_at: 1.hour.ago
+        create :user, name: 'new_user_4', created_at: 1.hour.ago
+        create :user, name: 'new_user_5', created_at: 4.hour.ago
+
+        get '/api/v1/users', created_at: 1.day.ago, api_key: api_token.token
+      end
+
+      it 'returns 2 users' do
+        users_name = JSON.parse(response.body).map{ |u| u['name'] } 
```

Comment:
What do you think of moving this to a `subject`?
```Ruby
subject { JSON.parse(response.body).map(&:name) }
```

Then, here, you can just do:
```Ruby
it ... do
  is_expected.to contain_exactly('new_user_3', 'new_user_4', 'new_user_5')
end
```